### PR TITLE
Add Intel Parallel Studio 2016, update 1

### PR DESCRIPTION
**Pull request by jas02**
Updated version of Intel Parallel Studio 2016 packages.

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/i/icc/icc-2016.1.150-GCC-4.9.3.eb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+name = 'icc'
+version = '2016.1.150'
+
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "C and C++ compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['parallel_studio_xe_%(version_major)s_composer_edition_for_cpp_%(version)s.tgz']
```
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.1.150-GCC-4.9.3.eb]: wait, the tarball still only has `2016`, the same as the previous release?

please tell me that's not the case...

Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.1.150-GCC-4.9.3.eb]: ah, no, you have both `%(version_major)s` and `%(version)s` in there, nevermind :)

File: easybuild/easyconfigs/i/intel/intel-2016.00.eb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+name = 'intel'
+version = '2016.00'
+
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolkit Compiler Edition provides Intel C/C++ and Fortran compilers, Intel MPI & Intel MKL."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+compver = '2016.1.150'
+gccsuff = '-GCC-4.9.3'
```
Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: usually, the `intel` toolchains are `GNU` based, but we're redefining `GNU` and `GCC` in other PRs so that binutils is included in both cases, so I need to take another look at that to have a full pictures here

@ocaisa: thoughts?

**Comment by boegel** [comment]
@jas02: please add an HPL easyconfig that uses this new toolchain?

File: easybuild/easyconfigs/i/intel/intel-2016.00.eb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
+easyblock = "Toolchain"
+
+name = 'intel'
+version = '2016.00'
```
Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: hmm, not sure if this is the best version, maybe we should stick to `2016.1.150`?

that is: the first update of the 2016 version of these tools; the `150` part comes back in the `impi` and `imkl` versions to indicate that all of these 'match'

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 7 (7 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/01393f7b95e1fc075220 for a full test report.

File: easybuild/easyconfigs/i/intel/intel-2016.00.eb (hunk 3 above)
Comment (jas02) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: `impi` is in version 5.1.2.150 (update 2). Other components of intel toolchain are update1.  So, there is not consistency from Intel side.

This is just maintenance release, changing every time release of toolchain is not what our users want, I think. They want to use latest updated versions of all Intel tools, but in familiar toolchain name (intel-2016.00). What do you think?

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 7 (7 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c54a0911f990136510bf for a full test report.

File: easybuild/easyconfigs/i/intel/intel-2016.00.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: Hmm, I see your point. The `.00` just seems weird to me, what does that mean? initial release?

I don't feel really strongly about it, just as long as we can motivate whatever we come up with

Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: Ah, but this isn't the initial release, this is update 1, so it should be `2016.01`?

Comment (jas02) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: I used convention from #1973. You had no objections against naming convention and I thought that it will be new toolchain name.

Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: well, except that #1973 deals with icc/ifort 2016.0.109, while here you have 2016.1.150... The bump in the compiler version should be reflected in the toolchain version, no?

Now both this PR and #1973 propose to use `2016.00`... We want to support installing both the 2016.0.109 and 2016.1.150 versions (for benchmarking/testing, for example).

**Comment by wpoely86** [comment]
@jas02 do these work? You need changes in the easyblock?

I'm gonna try to fixes the easyblock using hpcugent/easybuild-easyblocks#691 and your current branch.

File: easybuild/easyconfigs/i/intel/intel-2016.00.eb (hunk 3 above)
Comment (jas02) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: OK, how to solve it? Should I rename version to `2016.01`? And every other "major" update will be increment of this number? We have to be consistent in the future.

Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2016.00.eb]: Using `2016.01` makes sense to me, but we can discuss this during the conf call this week (I'll fix the date/time for that ASAP).

File: easybuild/easyconfigs/i/icc/icc-2016.1.150-GCC-4.9.3.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2016.1.150-GCC-4.9.3.eb]:
```python
sources = ['parallel_studio_xe_%(version_major)s_composer_edition_for_cpp_update%(version_minor)s.tgz']
```

File: easybuild/easyconfigs/i/imkl/imkl-11.3.1.150-iimpi-2016.00-GCC-4.9.3.eb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+moduleclass = 'numlib'
+
+postinstallcmds = [
+    # extract the examples
+    'tar xvzf %(installdir)s/mkl/examples/examples_cluster.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_core_c.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_core_f.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_f95.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_mic.tgz -C %(installdir)s/mkl/examples/'
```
Comment (boegel) [inline easybuild/easyconfigs/i/imkl/imkl-11.3.1.150-iimpi-2016.00-GCC-4.9.3.eb]: hmm, it would be better to handle this via the imkl easyblock (do-once-and-forget)? thoughts @jas02?

File: easybuild/easyconfigs/i/impi/impi-5.1.2.150-iccifort-2016.1.150-GCC-4.9.3.eb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+moduleclass = 'mpi'
+
+postinstallcmds = [
+    'ln -s %(installdir)s/lib64/libmpi.so  %(installdir)s/lib64/libmpich.so',
+    'ln -s %(installdir)s/lib64/libmpigc4.so  %(installdir)s/lib64/libmpichcxx.so',
+    'ln -s %(installdir)s/lib64/libmpigf.so %(installdir)s/lib64/libfmpich.so',
+    'ln -s %(installdir)s/lib64/libmpigf.so %(installdir)s/lib64/libmpichf90.so',
+    'ln -s %(installdir)s/lib64/libmpi.so %(installdir)s/lib64/libmpl.so',
+    'ln -s %(installdir)s/lib64/libmpi.so %(installdir)s/lib64/libopa.so'
```
Comment (boegel) [inline easybuild/easyconfigs/i/impi/impi-5.1.2.150-iccifort-2016.1.150-GCC-4.9.3.eb]: @jas02: why is this needed? this should also be handled via the easyblock (where we can check on things, for example make sure that the target doesn't exist yet, etc.)?

**Comment by boegel** [comment]
@jas02: please merge https://github.com/jas02/easybuild-easyconfigs/pull/2 to fix (part of) the remarks

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
will be merged via #2219
